sale_timesheet_line_exclude/models/account_analytic_line.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2018-2019 Brainbean Apps
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = 'account.analytic.line'
+
+    exclude_from_sale_order = fields.Boolean(
+        string='Exclude from Sale Order',
+        help=(
+            'Checking this would exclude this timesheet entry from Sale Order'
+        ),
+    )
+
+    @api.onchange('task_id', 'employee_id')
```

Comment:
Why not add here the other condition `exclude_from_sale_order` as you are doing the same code inside?

https://github.com/OCA/sale-workflow/pull/834/files#diff-6b07d3f063d79d29a8834694cdbbf559R23

Reply:
Good point, I'm not sure how it's better to proceed herecc: @pedrobaeza

Reply:
I don't see why you need this onchange

Reply:
To override parent class behaviour?

Reply:
@pedrobaeza any feedback on if to join the methods? I still think it's better to keep overriding an overriding

Reply:
But still don't understand why you need to override this behavior. Please always that you add controversial code, you must document it with docstring or comment in code.

Reply:
```

Reply:
added description

Reply:
OK, understood, and my final call is this code:
```python
if self.project_id:
    return self._onchange_exclude_from_sale_order()
return super()._onchange_task_id_employee_id()
```

Reply:
Did that a bit differently - calling one onchange from other isn't good style :P

Reply:
That's not true. It's the most verbose way of doing it. You repeat an `if`, but not very important. Doing your way, I even prefer to save the extra method and simply add the other line.

Reply:
And potentially trigger unwanted behaviour is that method is ever overriden - why to make extra links between code where they are not justified?

Reply:
Inheritance is for being respected, so if you override in other module that onchange (which is specific for this task), then you surely want the same to be executed in this case.

Reply:
Adjusted the code